specification/dartLangSpec.tex

```diff
@@ -1479,8 +1480,8 @@ \subsubsection{Optional Formals}
 \begin{grammar}
 <defaultFormalParameter> ::= <normalFormalParameter> (`=' <expression>)?
 
-<defaultNamedParameter> ::= <normalFormalParameter> (`=' <expression>)?
-  \alt <normalFormalParameter> ( `:' <expression>)?
+<defaultNamedParameter> ::= \REQUIRED{}? <normalFormalParameter> (`=' <expression>)?
+  \alt \REQUIRED{}? <normalFormalParameter> ( `:' <expression>)?
```

Comment:
What about a non nullable named parameter, that has no default value, is this inferred as being required? 

I hope so because that would mean that we can get rid of allmost all cases of required.

Reply:
@eernstg yes.

@kasperpeulen when we discussed this, the feeling was that it was error prone that cutting and pasting a method signature to a type would change its meaning:

```
int foo({int x}) => x; // Implicitly required
int Function({int x}) f = foo; // This is an error, since the parameter in the function type is optional
```

Reply:
@eernstg this is already marked as a semantic error below.